src/thunderbots/software/test/ai/grsim_communication/dribble_primitive.cpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * This file contains the unit tests for the grsim command implementation
+ * of the DribblePrimitive class
+ */
+
+#include <gtest/gtest.h>
+#include <string.h>
+
+#include "ai/world/robot.h"
+#include "grsim_communication/grsim_command_primitive_visitor.h"
+
+TEST(GrsimCommandPrimitiveVisitorTest, visit_dribble_primitive_kick_allowed)
+{
+    DribblePrimitive* dribble_primitive =
```

Comment:
There's no reason for this variable to be a pointer, and we _definitely_ shouldn't use raw pointers. In general it's good practice to only use pointers when necessary, and use smart-pointers instead of raw pointers. This goes for the entire file.